VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -88,73 +52,72 @@ public static void setAuthorAttributionText(@NonNull TextView textView,
         final CharSequence text;
         {
             final Context context = textView.getContext();
+            List<CharSequence> joinableStrings = new ArrayList<>();
+            boolean isEndorsed = false;
+            switch (authorAttributionLabel) {
+                case ANSWER:
+                    isEndorsed = true;
+                    joinableStrings.add(context.getString(R.string.discussion_post_marked_as_answer));
+                    break;
+                case ENDORSEMENT:
+                    isEndorsed = true;
+                    joinableStrings.add(context.getString(R.string.discussion_post_endorsed));
+                    break;
+            }
+
             final Date dateCreated = authorData.getCreatedAt();
-            final CharSequence formattedTime = dateCreated == null ? null :
-                    getRelativeTimeSpanString(context, initialTimeStampMs, dateCreated.getTime());
+            if (dateCreated != null) {
+                joinableStrings.add(getRelativeTimeSpanString(context, initialTimeStampMs,
+                        dateCreated.getTime()));
+            }
 
             final String author = authorData.getAuthor();
-            if (TextUtils.isEmpty(author)) {
-                switch (authorAttributionLabel) {
-                    case POST:
-                        text = formattedTime;
-                        break;
-                    default:
-                        text = null;
-                        break;
-                }
-            } else {
-                final String authorLabel = authorData.getAuthorLabel();
-
+            if (!TextUtils.isEmpty(author)) {
                 final SpannableString authorSpan = new SpannableString(author);
                 if (config.isUserProfilesEnabled() && !authorData.isAuthorAnonymous()) {
-                    authorSpan.setSpan(new ClickableSpan() {
-                        @Override
-                        public void onClick(View widget) {
-                            onAuthorClickListener.run();
-                        }
+                    // Changing the author text color and style
+                    authorSpan.setSpan(new ForegroundColorSpan(
+                                    context.getResources().getColor(R.color.edx_brand_primary_base)),
+                            0, authorSpan.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+                    authorSpan.setSpan(new StyleSpan(Typeface.BOLD), 0, authorSpan.length(),
+                            Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
 
+                    // Setting the click listener on the whole textView
+                    textView.setOnClickListener(new View.OnClickListener() {
```

Comment:
The click listener also needs to be set to null if the author data is not available, to handle view recycling in `ListView` etc.